src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -205,6 +206,24 @@ private async Task InitializeAsync()
             var solutionLoadedGuid = VSConstants.UICONTEXT.SolutionExistsAndFullyLoaded_guid;
             _vsMonitorSelection.GetCmdUIContextCookie(ref solutionLoadedGuid, out _solutionLoadedUICookie);
 
+            if (ErrorHandler.Succeeded(_vsSolution.GetProperty((int)__VSPROPID.VSPROPID_IsSolutionOpen, out object isSoltuionOpen)))
+            {
+                _isSolutionOpen = (bool)isSoltuionOpen;
+            }
+            else
+            {
+                _isSolutionOpen = false;
+            }
+
+            if (ErrorHandler.Succeeded(_vsSolution.GetProperty((int)__VSPROPID.VSPROPID_SolutionDirectory, out object solutionDirectory)))
+            {
+                _solutionDirectory = (string)solutionDirectory;
+            }
+            else
+            {
+                _solutionDirectory = null;
+            }
```

Comment:
How about assigning `_solutionDirectory to null` above the if statement so that we can skip the else block?

Reply:
Resolving as per https://github.com/NuGet/NuGet.Client/pull/4879/files/fd029a7386b50328354b64c2e9d603712aa542ed#r1044708022